tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,1338 @@
+      &::GeneralizedHarmonic::DampedHarmonicGauge_detail::
+          time_deriv_of_roll_on_function,
+      "DampedHarmonicGaugeQuantities", "time_deriv_roll_on_function",
+      {{{std::numeric_limits<double>::denorm_min(), 10.}}}, used_for_size);
+  // log_factor
+  pypp::check_with_random_values<1>(
+      static_cast<Scalar<DataType> (*)(const Scalar<DataType>&,
+                                       const Scalar<DataType>&, const double)>(
+          &::GeneralizedHarmonic::DampedHarmonicGauge_detail::log_factor<
+              DataType>),
+      "DampedHarmonicGaugeQuantities", "log_fac",
+      {{{std::numeric_limits<double>::denorm_min(), 10.}}}, used_for_size);
+  // spacetime_deriv_of_log_factor
+  pypp::check_with_random_values<1>(
+      static_cast<tnsr::a<DataType, SpatialDim, Frame> (*)(
+          const Scalar<DataType>&, const tnsr::I<DataType, SpatialDim, Frame>&,
+          const tnsr::A<DataType, SpatialDim, Frame>&,
+          const tnsr::II<DataType, SpatialDim, Frame>&, const Scalar<DataType>&,
+          const tnsr::ii<DataType, SpatialDim, Frame>&,
+          const tnsr::aa<DataType, SpatialDim, Frame>&,
+          const tnsr::iaa<DataType, SpatialDim, Frame>&, const double)>(
+          &::GeneralizedHarmonic::DampedHarmonicGauge_detail::
+              spacetime_deriv_of_log_factor<SpatialDim, Frame, DataType>),
+      "DampedHarmonicGaugeQuantities", "spacetime_deriv_log_fac",
+      {{{std::numeric_limits<double>::denorm_min(), 10.}}}, used_for_size);
+  // spacetime_deriv_of_power_log_factor
+  pypp::check_with_random_values<1>(
+      &wrap_spacetime_deriv_of_power_log_factor<SpatialDim, Frame, DataType>,
+      "DampedHarmonicGaugeQuantities", "spacetime_deriv_pow_log_fac",
+      {{{std::numeric_limits<double>::denorm_min(), 10.}}}, used_for_size,
+      1.e-11);
+}
+
+// Test gauge source function with randomized input
+template <size_t SpatialDim, typename Frame>
+void test_damped_harmonic_h_function(const DataVector& used_for_size) noexcept {
+  // H_a
+  pypp::check_with_random_values<1>(
+      static_cast<typename db::item_type<
+          GeneralizedHarmonic::Tags::GaugeH<SpatialDim, Frame>> (*)(
+          const typename db::item_type<
+              GeneralizedHarmonic::Tags::InitialGaugeH<SpatialDim, Frame>>&,
+          const Scalar<DataVector>&,
+          const tnsr::I<DataVector, SpatialDim, Frame>&,
+          const Scalar<DataVector>&,
+          const tnsr::aa<DataVector, SpatialDim, Frame>&, const double,
+          const double, const double,
+          const tnsr::I<DataVector, SpatialDim, Frame>&, const double)>(
+          &::GeneralizedHarmonic::DampedHarmonicHCompute<SpatialDim,
+                                                         Frame>::function),
+      "DampedHarmonicGaugeQuantities", "damped_harmonic_gauge_source_function",
+      {{{std::numeric_limits<double>::denorm_min(), 10.}}}, used_for_size);
+}
+
+// Test term 1/4 of gauge source function for Kerr-Schild
```

Comment:
Please remove the Kerr-Schild solution from these test functions. You are testing the pointwise functions, which you can do with random values. Just `make_with_random_values` the GR quantities. The specialisation to a Kerr-Schild solution doesn't add anything to these tests. This also means you can remove the code that sets up an LGL mesh, coordinate maps and such.

Reply:
I removed KerrSchild solution from these tests.

In addition, I added some 'analytic' tests for both gauge source function H and its spacetime derivatives for Schwarzschild spacetime. In these, I check for the equality of the output from `DampedHarmonicHCompute` and `SpacetimeDerivDampedHarmonicHCompute` `ComputeTag`'s, against ab-initio construction of both quantities for Schwarzschild spacetime in Kerr-Schild coordinates. The choice of coordinates is arbitrary, so I thought KS are as good as any.

Reply:
The setup of the LGL mesh should remain, since some coordinates are required e.g. by the `weight_function`. One could probably set up a random mesh, but the current implementation is sufficient.